storage/AggregationOperationState.hpp

```diff
@@ -156,6 +152,29 @@ class AggregationOperationState {
       const CatalogDatabaseLite &database);
 
   /**
+   * @brief Get the number of partitions to be used for initializing the
+   *        aggregation.
+   *
+   * @return The number of partitions to be used for initializing the aggregation.
+   **/
+  std::size_t getNumInitializationPartitions() const;
+
+  /**
+   * @brief Get the number of partitions to be used for finalizing the
+   *        aggregation.
+   *
+   * @return The number of partitions to be used for finalizing the aggregation.
+   **/
+  std::size_t getNumFinalizationPartitions() const;
```

Comment:
In the distributed scenario, you may need the table to be "really partitioned" (i.e. divided into several tables) in a different sense of what partitions refer to here. The partitions here cannot be scheduled to difference nodes since they have to update the same table.

So later when we make `CollisionFreeVectorTable` support really partitioned input relations, we will can rename all the "partitions" here to some other name, say `segments`.